compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -1112,8 +1125,48 @@ pub fn typeid_for_instance<'tcx>(
     mut instance: Instance<'tcx>,
     options: TypeIdOptions,
 ) -> String {
-    if matches!(instance.def, ty::InstanceDef::Virtual(..)) {
-        instance.args = strip_receiver_auto(tcx, instance.args)
+    if (matches!(instance.def, ty::InstanceDef::Virtual(..))
+        && instance.def_id() == tcx.lang_items().drop_in_place_fn().unwrap())
+        || matches!(instance.def, ty::InstanceDef::DropGlue(_, Some(_ty)))
```

Comment:
This is wrong for trivial drops. I adjusted your `cfi-drop-in-place.rs` test to include:
```
struct CustomDrop;                                                                         
impl Drop for CustomDrop {                                                                 
    fn drop(&mut self) {}                                                                  
}                                                                                          
                                                                                           
struct TrivialDrop;                                                                        
                                                                                           
fn main() {                                                                                
    let _ = Box::new(CustomDrop) as Box<dyn Send>;                                         
    let _ = Box::new(TrivialDrop) as Box<dyn Send>;                                        
}  
```
and the second drop SIGILL's.

This happens because the `Virtual` callsite cannot tell whether the drop glue is trivial, but your definition site is only abstracting nontrivial drop glue.

Reply:
Reverted to the original implementation from #116404. Lets have this working and we can try to improve/reduce it later and send another PR.